### ISSUE TITLE
Permission denied

### ISSUE THREAD

**Issue by sanmadjack** [PhasecoreX/docker-red-discordbot#25]
**Describe the bug**
When I try to run the docker image the container stops at this point:

Successfully built PyYAML
Installing collected packages: attrs, colorama, apsw-wheels, typing-extensions, aiosqlite, python-Levenshtein-wheels, multidict, chardet, async-timeout, idna, yarl, aiohttp, discord.py, contextlib2, schema, tqdm, click, pytz, aiohttp-json-rpc, appdirs, markdown, PyYAML, Red-Lavalink, distro, Babel, uvloop, fuzzywuzzy, Red-DiscordBot
Successfully installed Babel-2.8.0 PyYAML-5.3.1 Red-DiscordBot-3.4.3 Red-Lavalink-0.7.1 aiohttp-3.6.2 aiohttp-json-rpc-0.13.2 aiosqlite-0.15.0 appdirs-1.4.4 apsw-wheels-3.30.1.post3 async-timeout-3.0.1 attrs-19.3.0 chardet-3.0.4 click-7.1.2 colorama-0.4.3 contextlib2-0.6.0.post1 discord.py-1.5.1 distro-1.5.0 fuzzywuzzy-0.18.0 idna-2.10 markdown-3.2.2 multidict-4.7.6 python-Levenshtein-wheels-0.13.1 pytz-2020.1 schema-0.7.2 tqdm-4.48.0 typing-extensions-3.7.4.2 uvloop-0.14.0 yarl-1.5.1
Setting bot token...
/app/start-redbot.sh: 57: /app/functions/main-loop.sh: redbot: Permission denied

**Run command**
docker run -v red-discordbot:/data -e TOKEN=token -e PREFIX=.   phasecorex/red-discordbot

**Environment info:**
Arch Linux, up-to-date as of last monday
Docker version 19.03.13-ce, build 4484c46d9d

**Additional context**

**Comment by sanmadjack** [comment]
After reading #17 I created a folder in /opt/ owned by user 100 and it all worked. The issue seems to be happening due to using a docker volume, which is stored in /var/lib/docker/volumes and has parents blocked by everyone but root. I think this may be a source of the issue.

**Comment by PhasecoreX** [comment]
Yeah I still don't understand why that would happen... I can start up a container and have the `/data` mount be a docker volume and it works just fine. `/var/lib/docker/volumes` is only accessible to root as well (I think that's how it's supposed to be anyway).

Do you have anything like AppArmor/SELinux/Tomoyo running on the host that might be preventing executing files? I'm glad you at least got a workaround for the time being.

**Comment by agilly1989** [comment]
Yea, I am having the same issue, tried everything I could from #17 and this one and still getting the Permission Denied. No app armour or anything from what I can see.

`Activating Python virtual environment...
Updating Red-DiscordBot...
Requirement already satisfied: pip in /data/venv/lib/python3.8/site-packages (20.3)
Requirement already satisfied: setuptools in /data/venv/lib/python3.8/site-packages (50.3.2)
Requirement already satisfied: wheel in /data/venv/lib/python3.8/site-packages (0.36.0)
Requirement already satisfied: Red-DiscordBot in /data/venv/lib/python3.8/site-packages (3.4.3)
Requirement already satisfied: tqdm==4.48.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (4.48.0)
Requirement already satisfied: aiosqlite==0.15.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.15.0)
Requirement already satisfied: markdown==3.2.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.2.2)
Requirement already satisfied: contextlib2==0.6.0.post1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.6.0.post1)
Requirement already satisfied: click==7.1.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (7.1.2)
Requirement already satisfied: fuzzywuzzy==0.18.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.18.0)
Requirement already satisfied: appdirs==1.4.4 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (1.4.4)
Requirement already satisfied: aiohttp==3.6.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.6.2)
Requirement already satisfied: Red-Lavalink==0.7.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.7.1)
Requirement already satisfied: python-Levenshtein-wheels==0.13.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.13.1)
Requirement already satisfied: aiohttp-json-rpc==0.13.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.13.2)
Requirement already satisfied: PyYAML==5.3.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (5.3.1)
Requirement already satisfied: multidict==4.7.6 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (4.7.6)
Requirement already satisfied: uvloop==0.14.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.14.0)
Requirement already satisfied: attrs==19.3.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (19.3.0)
Requirement already satisfied: distro==1.5.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (1.5.0)
Requirement already satisfied: async-timeout==3.0.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.0.1)
Requirement already satisfied: discord.py==1.5.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (1.5.1)
Requirement already satisfied: pytz==2020.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (2020.1)
Requirement already satisfied: Babel==2.8.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (2.8.0)
Requirement already satisfied: yarl==1.5.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (1.5.1)
Requirement already satisfied: typing-extensions==3.7.4.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.7.4.2)
Requirement already satisfied: idna==2.10 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (2.10)
Requirement already satisfied: apsw-wheels==3.30.1.post3 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.30.1.post3)
Requirement already satisfied: schema==0.7.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.7.2)
Requirement already satisfied: chardet==3.0.4 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.0.4)
Requirement already satisfied: colorama==0.4.3 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.4.3)
Requirement already satisfied: async-timeout==3.0.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.0.1)
Requirement already satisfied: yarl==1.5.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (1.5.1)
Requirement already satisfied: chardet==3.0.4 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.0.4)
Requirement already satisfied: attrs==19.3.0 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (19.3.0)
Requirement already satisfied: multidict==4.7.6 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (4.7.6)
Requirement already satisfied: aiohttp==3.6.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.6.2)
Requirement already satisfied: typing-extensions==3.7.4.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.7.4.2)
Requirement already satisfied: pytz==2020.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (2020.1)
Requirement already satisfied: aiohttp==3.6.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.6.2)
Requirement already satisfied: aiohttp==3.6.2 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (3.6.2)
Requirement already satisfied: discord.py==1.5.1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (1.5.1)
Requirement already satisfied: contextlib2==0.6.0.post1 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (0.6.0.post1)
Requirement already satisfied: idna==2.10 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (2.10)
Requirement already satisfied: multidict==4.7.6 in /data/venv/lib/python3.8/site-packages (from Red-DiscordBot) (4.7.6)
Starting Red-DiscordBot!
/app/functions/main-loop.sh: line 69: /data/venv/bin/redbot: Permission denied`

**Comment by PhasecoreX** [comment]
@agilly1989 are you mounting a non-linux hard drive/partition at `/srv/dev-disk-by-label-500GB/`? That might be why permissions are all messed up for you, if for example the hard drive is using ntfs or something. I could also be 100% wrong, so take this with a grain of salt. Another issue here mentioned that the user couldn't get it to work when the bind mount to `/data` was on the network instead of local.

**Comment by agilly1989** [comment]
I do believe that I have that volume formatted to either ext3 or ext4 (will confirm when I get home). Host OS is OMV (built on Debian). It is a local drive (so, not networked). Saying that, I have multiple linuxserver images using that drive without issues.

What I'll do is try a folder on the OS drive to see if that volume I am using could be the issue.

**Comment by PhasecoreX** [comment]
Before you do that, can you try pulling and running the most recent docker image I just built and pushed? I thought of a way that hopefully will fix your (and @sanmadjack's) issue.

**Comment by agilly1989** [comment]
Solved (on my instance anyway) @PhasecoreX, Thanks for that :D

**Comment by sanmadjack** [comment]
I'll update and test this.

**Comment by PhasecoreX** [comment]
So it turns out that how I solved this is more of a workaround than an actual fix, and the root cause is still present (see #26 for another issue that arose). Basically your `/data` volume is coming from a filesystem that is not executable (it has the `noexec` flag set).

I have reverted this workaround and have instead added a check on startup that makes sure the `/data` volume doesn't have the `noexec` flag set. If it does have the flag, a warning will show and the container will not start (since `redbot` wouldn't start anyway)
